src/main/java/org/magicdgs/readtools/tools/distmap/DistmapPartDownloader.java

```diff
@@ -104,13 +107,26 @@
 
     private SamReaderFactory getSamReaderFactory() {
         if (factory == null) {
+            // TODO - a method for set a Path reference sequence should be included in HTSJDK
+            // TODO - this should be changed to the Path setter in HTSJDK (https://github.com/samtools/htsjdk/pull/1005)
+            final CRAMReferenceSource source = new ReferenceSource(getReferencePath());
             factory = SamReaderFactory.makeDefault()
-                    .referenceSequence(referenceFile)
+                    .referenceSource(source)
                     .validationStringency(readValidationStringency);
         }
         return factory;
     }
 
+    // reference path constructed on demand
+    private Path referencePath = null;
+
+    private Path getReferencePath() {
+        if (referencePath == null && referenceFile != null) {
+            referencePath = IOUtils.getPath(referenceFile);
```

Comment:
Untested, we should add a command line with a reference (even if doing nothing).